src/hooks/useWatchPendingTxs.ts

```diff
@@ -1,158 +1,188 @@
+import { SupportedCurrencyKey } from '@/references';
+import { time } from '@/utils/time';
+import { getUniqueId } from '@/utils/ethereumUtils';
+import { usePositionsStore } from '@/state/positions/positions';
+import { useClaimablesStore } from '@/state/claimables/claimables';
 
-export const useWatchPendingTransactions = ({ address }: { address: string }) => {
-  const storePendingTransactions = usePendingTransactionsStore(state => state.pendingTransactions);
-  const setPendingTransactions = usePendingTransactionsStore(state => state.setPendingTransactions);
+const ASSET_DETECTION_TIMEOUT = time.seconds(30) / 1000;
+
+async function fetchTransaction({
+  address,
+  currency,
+  transaction,
+}: {
+  address: string;
+  currency: SupportedCurrencyKey;
+  transaction: RainbowTransaction;
+}): Promise<RainbowTransaction> {
+  try {
+    if (!transaction.chainId || !transaction.hash) {
+      throw new Error('Pending transaction missing chainId or hash');
+    }
+    const fetchedTransaction = await transactionFetchQuery({
+      address,
+      chainId: transaction.chainId,
+      currency,
+      hash: transaction.hash,
+      originalType: transaction.type,
+    });
 
-  const pendingTransactions = useMemo(() => storePendingTransactions[address] || [], [address, storePendingTransactions]);
+    return {
+      ...transaction,
+      ...fetchedTransaction,
+    };
+  } catch (e) {
+    logger.error(new RainbowError('[fetchTransaction]: Failed to fetch transaction', e), {
+      transaction,
+    });
+  }
+
+  return transaction;
+}
 
+async function refetchUserAssets({ address }: { address: string }) {
+  await Promise.all([
+    userAssetsStore.getState().fetch(undefined, { force: true }),
+    usePositionsStore.getState().fetch(undefined, { force: true }),
+    useClaimablesStore.getState().fetch(undefined, { force: true }),
+    invalidateAddressNftsQueries(address),
+  ]);
+}
+
+export const useWatchPendingTransactions = ({ address }: { address: string }) => {
   const nativeCurrency = userAssetsStoreManager(state => state.currency);
+  const pendingTransactionsByAddress = usePendingTransactionsStore(state => state.pendingTransactions);
+  const pendingTransactions = useMemo(() => pendingTransactionsByAddress[address] || [], [address, pendingTransactionsByAddress]);
+
+  // Mined transactions we're waiting for asset changes for
+  const [waitingMinedTransactions, setWaitingMinedTransactions] = useState<MinedTransaction[]>([]);
 
-  const refreshAssets = useCallback(() => {
-    userAssetsStore.getState().fetch(undefined, { force: true });
-    invalidateAddressNftsQueries(address);
-  }, [address]);
+  const watchPendingTransactions = useCallback(async () => {
+    if (!pendingTransactions.length && !waitingMinedTransactions.length) return;
+
+    let newlyMinedTransactions: MinedTransaction[] = [];
+
+    if (pendingTransactions.length) {
+      const fetchedTransactions = await Promise.all(
```

Comment:
Out of curiosity, why do we need to fetch pending transactions differently from mined ones?